deep-learning/_posts/2018-02-15-RNN_Reinforce.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: review
+title:  "Neural Architecture Search with Reinforcement Learning"
+tags:   deep-learning RNN LSTM CNN reinforcement essentials 
+author: Pierre-Marc Jodoin 
+pdf:   "https://arxiv.org/pdf/1611.01578.pdf"
+cite:
+  authors: "Barret Zoph, Quoc V.Le"
+  title:   "Neural Architecture Search with Reinforcement Learning"
+  venue:   "ICLR 2017"
+---
+
+<div align="middle">
+  <img src="/deep-learning/images/rnn_reinforce_arch_search/sc01.png" width="450">
+</div>
+
+
+
+The goal of this paper is to search for neural architectures.  Since the architecture of a network can be seen as a sequence of elements, they predict a network A with an **RNN controler** (or policy network).  Once A is generated, it is trained and its validation accuracy is used as **reward to train the controler** afterwards.  More specifically, the controler is executed $$m$$ times, thus generating $$m$$ architectures $$A_i$$ to $$A_m$$.  Then those $$m$$ networks are then trained and validated which results into $$m$$ validation accuracies (or rewards).  The system then uses **policy gradient** to update the controler.
+
+The actual policy gradient equation is as follows where $$T$$ is the number of hyperparameters the controller had to predict to design a neural network architecture and $$a_t$$ is the t-th element in the architecture 
+
+<div align="middle">
+  <img src="/deep-learning/images/rnn_reinforce_arch_search/sc08.png" width="250">
+</div>
+
+
+Here is an exemple of a recurrent controler generating the structure of a simple CNN (not clear what is input is though)
```

Comment:
The input is a random configuration 
https://github.com/titu1994/neural-architecture-search/blob/master/train.py#L63

Reply:
The initial state is random, but the inputs to the RNN seems to be a simple sequence of numbers, where each position in the sequence has a specific meaning that the RNN doesn't know about...

Reply:
I'm guessing the RNN needs lots of examples to understand the "meaning" of each position. Training must be quite long...